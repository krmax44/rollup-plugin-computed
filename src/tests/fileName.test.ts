import type { Computers } from '..';
import { roll } from './utils';

describe('fileName', () => {
	it('respects custom file names', async () => {
		expect.assertions(2);

		const computers: Computers = {
			test: {
				fileName: 'bar.js',
				fn() {
					return { it: 'works' };
				}
			}
		};

		const output = await roll('basic.js', computers, false);
		const file: any = output.find(c => c.fileName === 'bar.js');

		expect(output.length).toBe(2);
		expect(file.code).toMatchSnapshot();
	});
});
