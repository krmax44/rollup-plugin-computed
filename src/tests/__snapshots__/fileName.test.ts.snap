// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileName respects custom file names 1`] = `
"var test = {\\"it\\":\\"works\\"};

export default test;
"
`;
